### ISSUE TITLE
When I change start day of the week from Sunday to Monday dates are set incorrectly

### ISSUE THREAD

**Issue by xyitest2** [liamcain/obsidian-calendar-plugin#346]
### Describe the bug

Changing start day of the week from Sunday to Monday should affect only which day is displayed as first column, instead it is changing the date.

### Steps to reproduce

Just do the above

### Expected behavior

Setting any day to act as first column shouldn't affect dates.

### Screenshots
According to the plugin 21.04.2024 is Monday:

![image](https://github.com/liamcain/obsidian-calendar-plugin/assets/9169501/a69d07c4-a2ee-4eee-9f0b-11319df82f5c)

#### OS

Windows

#### Obsidian Version (e.g. v0.10.6)

v1.5.3

### Theme (if applicable):

Wasp

**Comment by DesertDwarf** [comment]
I'm on the [beta version](https://github.com/liamcain/obsidian-calendar-plugin/releases/tag/2.0.0-beta.2) and I do see the problem you describe. However, I noticed if I close the config and switch to another month and back, everything is fixed. I'm guessing the programming flow immediately switches the day name columns but doesn't re-render the days of the month until you switch to another month.

Does it behave the same way on the 1.5.3 version?

**Comment by xyitest2** [comment]
Yes, on 1.5.3 version it behaves the same. Workaround confirmed. Thx.
